woocommerce/woocommerce-android
# Create CrowdSignal survey views and navigation

Summary
==========
Fixes issue #2644, this PR delivers the following features:
- Trigger Crowdsignal Feedback Survey when `Could be better` button is clicked on the `FeedbackRequestCard`
- Crowdsignal Feedback Survey view using a WebView
- WebView lifecycle handling with loading until survey view is completely loaded and survey completion interception
- Feedback Completed view with Dark Mode support
- Feedback fragments navigation, back stack and exit handled through Navigation Controller
- Contact Us text button integrated with `Help & Support` view

Changelog
==========
| Description | Commit |
| ------------- | ------------- |
| Initiate Survey fragments implementation | d0716b4d3d8290da28bbed92fbf200a17fb513c5 |
| Integrate Survey fragments with the Navigation Controller declaration structure |  9f3272e28df6da4572918cacf342d9570b867f64 |
| Create interception of the WebView lifecycle to make possible the loading dialog presentation and removal | e08773639d83e2425390b3b889ec247b3909386f |
| Make JavaScript execution available inside the WebView in order to better fit CrowdSignal web navigation experience | e55f021173383a6f2fb4b90b1a2cb0a78aad8ab5 |
| Create rule to intercept URL query parameter interaction through request loading override to verify if CrowdSignal survey is complete | 440f3c8fb9fb4d07829646ddf236f581cb4570b7 |
| `FeedbackCompletedFragment` created | db0dbad1b423964cde5df9d49acc2c5ee32b673d |
| `FeedbackCompletedFragment` UI structure defined | 530b3c6ac825e3766c1ab9531fdf11df5b06f786 |
| Change Survey URL declaration to the `AppUrls` object | 002f7f15062b54fa471d9f3e6d7f2f9dc103396b |
| Make the Crowdsignal Feedback Survey navigation start available from the `DashboardFragment` view | 2398d88a241251b4a1ea4933a89465445185291f |

Screenshots
==========
Light Mode
------------
| Click on `Could be better` button | Wait for the survey loading | Interact with the survey until it's finished | Verify if the expected view is presented and is fully functional | 
| ------------- | ------------- | ------------- | ------------- |
| ![Screenshot_1597814080](https://user-images.githubusercontent.com/5920403/90594826-c2b80900-e1c1-11ea-8368-af5b36c23dff.png) | ![Screenshot_1597812682](https://user-images.githubusercontent.com/5920403/90594829-c5b2f980-e1c1-11ea-9fb4-c8e04118efe3.png) | ![Screenshot_1597891397](https://user-images.githubusercontent.com/5920403/90712144-bcd12f00-e278-11ea-9b6c-b1b014139d00.png) | ![Screenshot_1597812776](https://user-images.githubusercontent.com/5920403/90594835-c9df1700-e1c1-11ea-9b16-63bdb680ed8c.png) |




Dark Mode
------------
| Click on `Could be better` button | Wait for the survey loading | Interact with the survey until it's finished | Verify if the expected view is presented and is fully functional | 
| ------------- | ------------- | ------------- | ------------- |
| ![Screenshot_1597891515](https://user-images.githubusercontent.com/5920403/90711653-a70f3a00-e277-11ea-96d8-f094ca3a722d.png) | ![Screenshot_1597891046](https://user-images.githubusercontent.com/5920403/90711657-a8d8fd80-e277-11ea-97d5-79f0a99c178d.png) | ![Screenshot_1597891102](https://user-images.githubusercontent.com/5920403/90711659-aa0a2a80-e277-11ea-9d2b-a462a141dcd6.png) | ![Screenshot_1597891109](https://user-images.githubusercontent.com/5920403/90711662-ab3b5780-e277-11ea-84c9-83b23f977fc0.png) |

Tablet
------------
| Click on `Could be better` button | Wait for the survey loading | Interact with the survey until it's finished | Verify if the expected view is presented and is fully functional | 
| ------------- | ------------- | ------------- | ------------- |
| ![Screenshot_1597891681](https://user-images.githubusercontent.com/5920403/90711694-cb6b1680-e277-11ea-92e0-5c83e858fcc2.png) | ![Screenshot_1597891688](https://user-images.githubusercontent.com/5920403/90711703-cc9c4380-e277-11ea-8911-e44f05a40bd9.png) | ![Screenshot_1597891695](https://user-images.githubusercontent.com/5920403/90711705-ce660700-e277-11ea-991c-b0ca7388a3c0.png) | ![Screenshot_1597891708](https://user-images.githubusercontent.com/5920403/90711709-cf973400-e277-11ea-8dab-671f24a8aa11.png) |

How to Test
------------

1. Access the `My Store` section from the Woo app
2. Inside the `FeedbackRequestCard`, click in the `Could be better` button
3. Verify that the survey view is called, the loading dialog is visible until the survey is completely loaded
4. Interact with the survey until the conclusion, after finishing it, verify that the custom native Feedback Completed View is called instead of letting the WebView show the default web one.
5. Once inside the Feedback Completed View, verify that the each button is working as expected and described


Update release notes:
